docs/machine-learning/how-to-guides/explain-machine-learning-model-permutation-feature-importance-ml-net.md

```diff
@@ -0,0 +1,159 @@
+    [LoadColumn(1)]
+    public float ResidentialZones { get; set; }
+
+    [LoadColumn(2)]
+    public float CommercialZones { get; set; }
+
+    [LoadColumn(3)]
+    public float NearWater { get; set; }
+
+    [LoadColumn(4)]
+    public float ToxicWasteLevels { get; set; }
+
+    [LoadColumn(5)]
+    public float AverageRoomNumber { get; set; }
+
+    [LoadColumn(6)]
+    public float HomeAge { get; set; }
+
+    [LoadColumn(7)]
+    public float BusinessCenterDistance { get; set; }
+
+    [LoadColumn(8)]
+    public float HighwayAccess { get; set; }
+
+    [LoadColumn(9)]
+    public float TaxRate { get; set; }
+
+    [LoadColumn(10)]
+    public float StudentTeacherRatio { get; set; }
+
+    [LoadColumn(11)]
+    public float PercentPopulationBelowPoverty { get; set; }
+
+    [LoadColumn(12)]
+    [ColumnName("Label")]
+    public float Price { get; set; }
+}
+```
+
+Load the data into an [`IDataView`](xref:Microsoft.ML.IDataView).
+
+## Train the model
+
+The code sample below illustrates the process of training a linear regression model to predict house prices.
+
+```csharp
+// 1. Split data into training and test sets
+DataOperationsCatalog.TrainTestData dataSplit = mlContext.Data.TrainTestSplit(data);
+IDataView trainData = dataSplit.TrainSet;
+IDataView testData = dataSplit.TestSet;
+
+// 2. Get the column name of input features.
+string[] featureColumnNames = 
+    data.Schema
+        .Select(column => column.Name)
+        .Where(columnName => columnName != "Label").ToArray();
+
+// 3. Define estimator with data pre-processing steps
+IEstimator<ITransformer> dataPrepEstimator =
+    mlContext.Transforms.Concatenate("Features", featureColumnNames)
+        .Append(mlContext.Transforms.NormalizeMinMax("Features"));
+
+// 4. Create transformer using the data pre-processing estimator
+ITransformer dataPrepTransformer = dataPrepEstimator.Fit(trainData);
+
+// 5. Pre-process the training and test sets
+IDataView preprocessedTrainData = dataPrepTransformer.Transform(trainData);
+IDataView preprocessedTestData = dataPrepTransformer.Transform(testData);
+
+// 6. Define Stochastic Dual Coordinate Ascent machine learning estimator
+var sdcaEstimator = mlContext.Regression.Trainers.Sdca();
+
+// 7. Train machine learning model
+var sdcaModel = sdcaEstimator.Fit(preprocessedTrainData);
+```
+
+## Explain the model with Permutation Feature Importance (PFI)
+
+Permutation Feature Importance (PFI) is a technique used to explain classification and regression models that is inspired by [Breiman's *Random Forests* paper](https://www.stat.berkeley.edu/~breiman/randomforest2001.pdf)(see section 10). At a high level, the way it works is by randomly shuffling data one feature at a time and calculating how much the performance metric of interest decreases. The larger the change, the more important that feature is.
+
+In ML.NET use the [`PermutationFeatureImportance`](xref:Microsoft.ML.PermutationFeatureImportanceExtensions) method for your respective task.
```

Comment:
This could be a little clearer.

"Append the PermutationFeatureImportance estimator to your pipeline" (if it is indeed an estimator)

Reply:
This step like DataOperationsCatalog operations are outside of the pipeline because it is not an estimator.

Reply:
#resolved